src/ui/components/Comments/TranscriptComments/CommentEditor/ExistingCommentEditor.tsx

```diff
@@ -59,7 +66,21 @@ function ExistingCommentEditor({
         }
       }}
     >
-      <CommentEditor editable={editable} comment={comment} handleSubmit={handleSubmit} />
+      <CommentEditor
+        editable={editable}
+        comment={comment}
+        handleSubmit={handleSubmit}
+        onCreate={({ editor }) => {
+          const storedComment = commentsLocalStorage.get();
+          if (storedComment) {
+            editor.commands.setContent(JSON.parse(storedComment));
+          }
+        }}
+        onUpdate={debounce(({ editor }) => {
+          commentsLocalStorage.set(JSON.stringify(editor.getJSON()));
+        }, PERSIST_COMM_DEBOUNCE_DELAY)}
```

Comment:
It looks like the onCreate and onUpdate logic are repeated in both `ExistingCommentEditor` and `NewCommentEditor` — is there a way to just pass the `mode` down and call the `useCommentsLocalStorage` at the `CommentEditor` component??

Reply:
Since we already have split components (`Exisiting` and `New`) to handle rendering and ~some logic, then it should be the responsibility of those components to provide their corresponding handlers. They're similar, but not quite the same.